### PR TITLE
Add Agentic-RAG notebooks with LlamaIndex

### PR DESCRIPTION
**Pull request by openvino-dev-samples**
TODO

- [ ] add gradio demo

### CODE REVIEW

File: notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb [hunk 1]
```diff
@@ -0,0 +1,597 @@
+{
```
Comment (eaidova) [inline notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb:1]: <p>maybe better to have this in cell, but commented like here <a href="https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/stable-diffusion-v3/stable-diffusion-v3.ipynb" target="_blank">https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/stable-diffusion-v3/stable-diffusion-v3.ipynb</a></p><p> </p><p>or if possible to use some model that does not required gated access</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2189/discussion/#comment-1677363922'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: .ci/skipped_notebooks.yml [hunk 2]
```diff
@@ -531,4 +531,10 @@
         - ubuntu-20.04
         - ubuntu-22.04
         - windows-2019
-
+- notebook: notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb
```
Comment (eaidova) [inline .ci/skipped_notebooks.yml:544]: @OpenVINO-dev-contest  is it possible to setup replacement with some smaller model for tests only? e.g. tiny-llama or qwen2 (we do not need to check generation quality, more about checking that notebook is functional

File: notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb [hunk 3]
```diff
@@ -0,0 +1,601 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6b19fb6e-e3f4-43e6-b7f1-5bbf2719fa85",
+   "metadata": {
+    "test_replace": {
+     "meta-llama/Meta-Llama-3-8B-Instruct": "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+    }
+   },
```
Comment (eaidova) [inline notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb]:
```suggestion
   "metadata": {
   },
```

File: notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb [hunk 4]
```diff
@@ -0,0 +1,601 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "86fdc4ba-74c4-4869-898e-131f47827e8f",
+   "metadata": {
+    "test_replace": {
+     "meta-llama/Meta-Llama-3-8B-Instruct": "Qwen/Qwen2-0.5B-Instruct"
+    }
```
Comment (eaidova) [inline notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb]:
```suggestion
"test_replace": {
     "meta-llama/Meta-Llama-3-8B-Instruct": "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
    }
```

File: notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+    "os.environ[\"GIT_CLONE_PROTECTION_ACTIVE\"] = \"false\"\n",
+    "\n",
+    "%pip install -Uq pip\n",
+    "%pip uninstall -q -y optimum optimum-intel\n",
+    "%pip install -q --extra-index-url https://download.pytorch.org/whl/cpu\\\n",
+    "\"llama-index\" \"pymupdf\" \"llama-index-readers-file\" \"llama-index-llms-openvino\" \"llama-index-embeddings-openvino\" \"transformers>=4.40\" \\\n",
+    "\"git+https://github.com/huggingface/optimum-intel.git\"\\\n",
+    "\"git+https://github.com/openvinotoolkit/nncf.git\"\\\n",
+    "\"datasets\"\\\n",
+    "\"accelerate\"\n",
```
Comment (eaidova) [inline notebooks/llm-agent-react/llm-agent-rag-llamaindex.ipynb]:
```suggestion
    "%pip install -q --extra-index-url https://download.pytorch.org/whl/cpu \\\n",
    "\"llama-index\" \"pymupdf\" \"llama-index-readers-file\" \"llama-index-llms-openvino\" \"llama-index-embeddings-openvino\" \"transformers>=4.40\" \\\n",
    "\"git+https://github.com/huggingface/optimum-intel.git\" \\\n",
    "\"git+https://github.com/openvinotoolkit/nncf.git\" \\\n",
    "\"datasets\" \\\n",
    "\"accelerate\"\n",
```

**Comment by openvino-dev-samples** [comment]
no idea why CI failed on Llama_index import, cuz it works fine with my clean environment.

**Comment by eaidova** [comment]
@OpenVINO-dev-contest which python version do you use for testing? I have suspicious that some of packages can be non available for python3.8 (deprecated version, some packages already exclude it from supported)

Could you please try to add it into ignored for python3.8 only?

**Comment by openvino-dev-samples** [comment]
> @OpenVINO-dev-contest which python version do you use for testing? I have suspicious that some of packages can be non available for python3.8 (deprecated version, some packages already exclude it from supported)
> 
> Could you please try to add it into ignored for python3.8 only?

Python 3.11

**Review by eaidova — APPROVED** [review]

**Pull request merged**